client/src/components/RocketProfilePopup.tsx

```diff
@@ -1,113 +1,211 @@
+	isOpen: boolean;
+	onSave: () => void;
+	onClose: () => void;
 }
 
-const cardStyle = {
-  position: 'absolute',
-  top: '50%',
-  left: '50%',
-  transform: 'translate(-50%, -50%)',
-  minWidth: 700,
-  boxShadow: 24,
-};
-
-const buttonStyle = {
-  height: 45,
-  borderRadius: 6,
-  mt: 10,
-  alignSelf: 'center',
-  px: 3,
+interface IRocketDetails {
+	result: {
+		Name: string;
+		Motor: string;
+		MotorType: string;
+		Height: number;
+		Class: string;
+		Mass: number;
+	};
 }
 
 const RocketProfilePopup: React.FC<RocketProfileProps> = (props: RocketProfileProps) => {
+	const [name, setName] = useState<string>();
+	const [motor, setMotor] = useState<string>();
+	const [mass, setMass] = useState<number>();
+	const [motorType, setMotorType] = useState<string>('');
+	const [rocketClass, setRocketClass] = useState<string>('');
+	const [height, setHeight] = useState<number>();
+	// POST or PATCH data depending on if were creating a new rocket or editing a old one
+	const save = async () => {
+		const payload = {
+			Height: height,
+			Class: rocketClass,
+			Motor: motor,
+			MotorType: motorType,
+			Name: name,
+			Mass: mass
+		};
+		if (props.rocketProfileId) {
+			await axios.patch(`http://127.0.0.1:9090/rocket/${props.rocketProfileId}`, payload);
+		} else {
+			await axios.post(`http://127.0.0.1:9090/rocket`, payload);
+		}
+	};
 
-  const [rocketClass, setRocketClass] = useState('');
-  const [motorType, setMotorType] = useState('');
-
-  const saveProfile = () => {
-    console.log("Saving profile");
-    props.onClose();
-  }
-
-  return (
-    <Box sx={cardStyle}>
-      <Card>
-        <CardHeader title="New Rocket Profile"
-        action={
-          <IconButton aria-label="Close" onClick={props.onClose}>
-            <Close />
-          </IconButton>
-        }/>
-        <hr></hr>
-        <CardContent>
-          <FormControl fullWidth>
-          <Stack direction="column" spacing={2}>
-            <Stack direction="row" spacing={4}>
-              <TextField sx={{minWidth: "40%"}} id="profile-name" label="Profile Name" variant="outlined" />
-              <Button size="small" sx={buttonStyle} variant="contained" startIcon={<Add />} component="label">
-                <input hidden accept="*" type="file" />
-                SVG
-              </Button>
-              <Button size="small" sx={buttonStyle} variant="contained" startIcon={<Add />} component="label">
-                <input hidden accept="*" type="file" />
-                SIM DATA
-              </Button>
-            </Stack>
+	const [editMode, setEditMode] = useState<boolean>(props.rocketProfileId !== '-1');
 
-              <TextField sx={{minWidth: "100%"}} id="mass" label="Mass" variant="outlined"
-              InputProps={{
-                endAdornment: <InputAdornment position="start">kg</InputAdornment>,
-              }}
-              />
+	useEffect(() => {
+		setEditMode(props.rocketProfileId !== '')
+		// Reset State
+		setHeight(NaN);
```

Comment:
The browser console doesn't like the Nan and throws `received NaN for the `value` attribute. If this is expected, cast the value to a string at input` so we can either cast it like it asks, or just leave if you want.